### PR TITLE
Translate vec-into-iter.md

### PR DESCRIPTION
**Pull request by toku-sa-n**

### CODE REVIEW

File: src/vec-into-iter.md [hunk 1]
```diff
@@ -51,17 +105,21 @@ struct IntoIter<T> {
...
 And this is what we end up with for initialization:
+-->
+
+そしてこれが初期化のコードです。
 
 ```rust,ignore
 impl<T> Vec<T> {
     fn into_iter(self) -> IntoIter<T> {
-        // Can't destructure Vec since it's Drop
+        // Vec がドロップされてしまうため、 Vec をデストラクト出来ません。
```
Comment (imsut) [inline src/vec-into-iter.md:117]: this means we cannot write
```
let Vec { ptr, cap, len } = self;
```
as `self` would drop.

[here](
https://doc.rust-jp.rs/book/second-edition/ch18-03-pattern-syntax.html#a%E6%A7%8B%E9%80%A0%E4%BD%93%E3%82%92%E5%88%86%E9%85%8D%E3%81%99%E3%82%8B) it calls it 分配

File: src/vec-into-iter.md [hunk 2]
```diff
@@ -1,46 +1,100 @@
...
+<!--
 Note that the order of read and offset are reversed for `next` and `next_back`
 For `next_back` the pointer is always after the element it wants to read next,
 while for `next` the pointer is always at the element it wants to read next.
 To see why this is, consider the case where every element but one has been
 yielded.
+-->
 
+読み出しとオフセットの操作の順序は `next` と `next_back` とで逆転することに
+注意してください。 `next_back` では、ポインタは次に読みたい要素の直前の
```
[the remark this answers is not in the document]
Comment (toku-sa-n) [reply src/vec-into-iter.md:60]: Ah, yes you're right.

File: src/vec-into-iter.md (hunk 1 above)
Comment (toku-sa-n) [reply src/vec-into-iter.md:117]: I didn't know the syntax (or I forgot about it). Thanks!

File: src/vec-into-iter.md [hunk 3]
```diff
@@ -123,16 +189,22 @@ impl<T> DoubleEndedIterator for IntoIter<T> {
...
 
+<!--
 Because IntoIter takes ownership of its allocation, it needs to implement Drop
 to free it. However it also wants to implement Drop to drop any elements it
 contains that weren't yielded.
+-->
+
+IntoIter はアロケーションの所有権を受け取るので、それを解放するために Drop を
+実装する必要があります。しかし、イテレーションの最中に返されなかった要素を
+ドロップするための Drop も実装する必要があります。
```
[the remark this answers is not in the document]
Comment (toku-sa-n) [reply src/vec-into-iter.md]: :+1:

**Pull request merged**